src/scenes/ApplicationRouter/NewApplicationModal.tsx

```diff
@@ -71,6 +71,15 @@ const NewApplicationModal = ({ open, onClose }: NewApplicationModalProps): JSX.E
     [availableProjects]
   );
 
+  useEffect(() => {
+    if (newApplication.projectType === 'New' && projectOptions.length) {
+      setNewApplication({
+        projectType: 'Existing',
+        projectId: undefined,
+      });
+    }
+  }, [newApplication, projectOptions, setNewApplication]);
+
```

Comment:
Since we can't determine if there are available projects without an application until `projectOptions` is computed, I set the default to `New` and added a `useEffect` to select `Existing` if any such projects are found.

I tried setting `projectId` to `projectOptions[0].value` to align with existing logic, but while `newApplication` updated, the Dropdown didn't reflect this change. Testing in Staging confirmed the Dropdown doesn't show the auto-selected `projectId`. Instead of spending more time on this, I set `projectId` to `undefined`. If auto-selecting in the Dropdown is necessary, I can revisit and modify that behavior.

Reply:
I think the issue is that the computation of the project options and the initialization of the form are happening in two different places. Also, the `useMemo` which is computing `projectOptions` is missing `allApplications` from its dependency array, so if `availableProjects` is done updating before the final value for `allApplications` is received, it will be incorrect. 

Generally these things should be done in a single `useEffect` statement, so you aren't juggling state values between multiple memoized functions and hoping the order of resolution is correct. I will push up a PR into this branch with the fix.

Reply:
Sounds good, thanks @nickgraz 🙌